edgeless_cli/src/edgeless_cli.rs

```diff
@@ -72,17 +72,26 @@ struct Args {
 #[derive(serde::Deserialize)]
 struct CLiConfig {
     controller_url: String,
-    pub repository_url: String,
-    pub repository_basic_auth_user: String,
-    pub repository_basic_auth_pass: String,
+    function_repository: Option<FunctionRepositoryConfig>,
+}
+
+#[derive(serde::Deserialize)]
+struct FunctionRepositoryConfig {
+    pub url: String,
+    pub basic_auth_user: String,
+    pub basic_auth_pass: String,
 }
 
 pub fn edgeless_cli_default_conf() -> String {
-    let controller_url = String::from("controller_url = \"http://127.0.0.1:7001\"");
-    let repository_url = String::from("repository_url = \"\"");
-    let repository_user = String::from("repository_basic_auth_user = \"\"");
-    let reposisitory_passwd = String::from("repository_basic_auth_pass = \"\"");
-    return format!("{}\n{}\n{}\n{}", controller_url, repository_url, repository_user, reposisitory_passwd);
+    String::from(
+        r##"controller_url = "http://127.0.0.1:7001"
+
+#[function_repository]
+#url = ""
```

Comment:
The old version used commented-out parts but was broken due to special characters in the comments.

Reply:
I missed that. This one should be fine.